Userland/Libraries/LibGL/SoftwareRasterizer.h

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2021, Stephan Unverwerth <s.unverwerth@gmx.de>
+ * 
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "GLStruct.h"
+#include <AK/Assertions.h>
+#include <AK/NonnullRefPtr.h>
+#include <LibGfx/Bitmap.h>
+
+static constexpr size_t RASTERIZER_BLOCK_SIZE = 16;
+
+//#define USE_SSE
+
+#ifdef USE_SSE
+#include <immintrin.h>
+#endif
+
+struct alignas(16) QFloat {
+    union {
+#ifdef USE_SSE
+        __m128 v;
+#endif
+        struct { float a, b, c, d; };
+    };
+
+    constexpr QFloat() = default;
+    constexpr QFloat(float a, float b, float c, float d) :a(a), b(b), c(c), d(d) {}
+#ifdef USE_SSE
+    constexpr QFloat(__m128 v) : v(v) {}
+#endif
+};
```

Comment:
I believe much of this is supported via `AK/SIMD.h`. According to ccapitalK, "They support all the normal c++ arithmetic operations, plus all the gcc x86 SSE builtins" so the compiler may do this for us if somebody compiles with `-msse`